doc/sphinx/proofs/automatic-tactics/auto.rst

```diff
@@ -313,27 +313,56 @@ Creating hint databases
 ```````````````````````
 
 Hint databases can be created with the :cmd:`Create HintDb` command or implicitly
-by adding a hint to an unknown database.  We recommend you always use :cmd:`Create HintDb`
-and then imediately use :cmd:`Hint Constants` and :cmd:`Hint Variables` to make
-those settings explicit.
-
-Note that the default transparency
-settings differ between these two methods of creation.  Databases created with
-:cmd:`Create HintDb` have the default setting `Transparent` for both `Variables`
-and `Constants`, while implicitly created databases have the `Opaque` setting.
+by adding a hint to an unknown database.  Databases created with
+:cmd:`Create HintDb` have the default setting `Transparent` (i.e. unfold all)
+for `Constants`, `Projections` and `Variables`, while implicitly created databases
+have the `Opaque` setting (i.e. unfold none).  We recommend using :cmd:`Hint Constants`,
+:cmd:`Hint Projections` and :cmd:`Hint Variables` immediately after :cmd:`Create HintDb`
+to make these settings explicit.
+
+Use :cmd:`Hint Opaque` and :cmd:`Hint Transparent` to control the opacity of
+individual items.  Hint opacity settings influence which hints the search tactics
+try, but they have no effect on how the selected tactic is executed.  The proof
```

Comment:
Definitely not. Hint opacity is also used by the unification algorithm of the underlying apply-like tactics.

Reply:
"apply-like tactics" meaning Hint Resolve?

`apply` ignores regular opaque, so I don't see your point.  (And Hint Extern doesn't unfold goal or hint in any case.) 

```
Definition one := 1.
Opaque one.
Theorem thm : 1 = one. reflexivity. Qed.

Goal one = 1.
Succeed apply thm.  (* ignores opaque setting *)
Fail unfold one.
```

Reply:
apply-like tactics do rely on the transparent state even though they ignore it in some places.

Reply:
Is that only for tc eauto, which uses `autoapply` or are there other mechanisms?